platform/darwin/src/MGLFeature.mm

```diff
@@ -177,6 +177,18 @@ - (NSDictionary *)geoJSONDictionary {
     return mbgl::Feature{geometry};
 }
 
+- (mbgl::FeatureCollection)mbglFeatureCollection {
+    mbgl::FeatureCollection featureCollection;
+    featureCollection.reserve(self.shapes.count);
+    for (id <MGLFeaturePrivate> feature in self.shapes) {
+        if ([feature isMemberOfClass:[MGLShapeCollectionFeature class]]) {
+            featureCollection.push_back([self mbglFeatureCollection]);
+        }
+        featureCollection.push_back([feature mbglFeature]);
```

Comment:
That was the intent of the if statement just above here. However I don't think we need to support this case (see [comment](https://github.com/mapbox/mapbox-gl-native/pull/7339/files#r91633052))